### PR TITLE
[feat] Add MonerisInterface and GatewayInterface

### PR DESCRIPTION
**Pull request by danielsdeboer**

### CODE REVIEW

File: src/Interfaces/GatewayInterface.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace CraigPaul\Moneris\Interfaces;
+
+interface GatewayInterface
+{
+    /**
+     * Capture a pre-authorized a transaction.
```
Comment (kaydenvanrijn) [inline src/Interfaces/GatewayInterface.php]: Mild typo

File: src/Interfaces/GatewayInterface.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     * @param mixed|null $amount
+     * @return \CraigPaul\Moneris\Response
+     */
+    public function capture ($transaction, $order = null, $amount = null);
+
+    /**
+     * Create a new Vault instance.
+     * @return \CraigPaul\Moneris\Vault
+     */
+    public function cards ();
```
Comment (kaydenvanrijn) [inline src/Interfaces/GatewayInterface.php:20]: Couldn't this just be called `vault ()` instead?

File: src/Interfaces/GatewayInterface.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Pre-authorize a purchase.
+     * @param array $params
+     * @return \CraigPaul\Moneris\Response
+     */
+    public function preauth (array $params = []);
```
Comment (kaydenvanrijn) [inline src/Interfaces/GatewayInterface.php:27]: Why not type hint like
`public function preauth (array $params = []): Response;`

File: src/Interfaces/GatewayInterface.php (hunk 2 above)
Comment (danielsdeboer) [reply src/Interfaces/GatewayInterface.php:20]: I don't want to actually change the API at all for BC considerations.

File: src/Interfaces/GatewayInterface.php (hunk 3 above)
Comment (danielsdeboer) [reply src/Interfaces/GatewayInterface.php:27]: I'm not adding any type hinting for now, just trying to ship something that works. I'm planning to change these over time but honestly the value proposition is kinda low for typehints at the moment.

**Review by kaydenvanrijn — APPROVED** [review]

**Pull request merged**